src/server.rs

```diff
@@ -1236,20 +1236,32 @@ impl proto::Peer for Peer {
         // Convert the URI
         let mut parts = uri::Parts::default();
 
-        if let Some(scheme) = pseudo.scheme {
-            let maybe_scheme = uri::Scheme::from_shared(scheme.clone().into_inner());
-            parts.scheme = Some(maybe_scheme.or_else(|why| malformed!(
-                "malformed headers: malformed scheme ({:?}): {}", scheme, why,
-            ))?);
-        } else {
-            malformed!("malformed headers: missing scheme");
-        }
 
+        // A request translated from HTTP/1 must not include the :authority
+        // header
```

Comment:
If a `host` header is present do we have any obligation to validate it?

Reply:
I can't find anything in the [HTTP2 spec](https://httpwg.org/specs/rfc7540.html) stating to do anything with a received `host` header besides handling it like any other.